README.md

```diff
@@ -52,102 +31,110 @@ import "github.com/roserocket/xerrs"
 //....
 
 if data, err := MyFunc(); err != nil {
-    err = xerrs.Extend(err) // extend error
+    err := xerrs.Extend(err) // extend error
 
     //....
 
-    xerrs.Stack(err) // print detailed error information including its stack
+    fmt.Println(err.Error())    // prints error
+
+    if x, ok := err.(*xerr); ok {
+        fmt.Println(x.Details()) // print detailed error information including its stack
+    } else {
+        fmt.Println(err.Error()) // print basic error message if it is original error
+    }
+
+    //....
 }
 ```
 
 ### Deferred logging + masking example
 
 ```go
+func ErrDetails(err error) string {
+    if x, ok := err.(*xerr); ok {
+        fmt.Println(x.Details()) // print detailed error information including its stack
+    } else {
+        fmt.Println(err.Error()) // print basic error message if it is original error
+    }
+}
+
 func DoSomething(w http.ResponseWriter, r *http.Request) {
     var err error
 
     defer func() {
-        xerrs.Stack(err)
+        ErrDetails(err)
     }()
 
     someModel := &Model{}
     err = ReadJSONFromReader(r, someModel)
     if err != nil {
         err = xerrs.Extend(err)
-        DoSomethingWithError(w, xerrs.Error(err))
+        DoSomethingWithError(w, err.Error()) // original error value is used here
         return
     }
 
     _, err = DBCreateMyModel(someModel)
     if err != nil {
         err = xerrs.MaskError(err, errors.New("We are experiencing technical difficulties"))
-        DoSomethingWithError(w, xerrs.Error(err))
+        DoSomethingWithError(w, err.Error()) // masked error value is used here. But original value would still be printed out in the defer
         return
     }
 
     OutputDataToClient(w, &someModel)
 }
 ```
 
-### Backward compatibility with existing code
+### Custom data in error
 
 ```go
-func VeryComplexLongFunction(arg1, arg2) error {
-    var err error
-    // convert error to an extended one and use it to for debugging purposes
+func ErrPrintDetails(err error) string {
+    if x, ok := err.(*xerr); ok {
+        fmt.Println(x.Details())
```

Comment:
i think this accepts an int.
also this could be a spot to comment that Details prints cause error, mask if specified, and stack